roles/neutron-common/templates/usr/local/bin/neutron-reset

```diff
@@ -53,6 +53,20 @@ class Interface(object):
 
         return cmd
 
+    @staticmethod
+    def get_neutron_interfaces(namespace=None):
+        return [x for x in Interface.enumerate(namespace) if x.is_neutron_interface()]
+
+    @staticmethod
+    def get_nova_interfaces(namespace=None):
+        return [x for x in Interface.enumerate(namespace) if x.is_nova_interface()]
+
+    @staticmethod
+    def get_vxlan_devices():
+        for iface in Interface.get_neutron_interfaces():
+            if 'vxlan' in iface.ifname:
+                yield iface
```

Comment:
This is a generator which uses a list which uses a generator. You might consider modifying get_neutron_interfaces() and get_nova_interfaces() to use this same pattern as a filtering generator.

Also, your naming conventions are inconsistent, this would be get_neutron_vxlan_interfaces().